znai-reactjs/src/doc-elements/latex/Latex.tsx

```diff
@@ -14,28 +14,33 @@
  * limitations under the License.
  */
 
-import React, { useEffect, useRef } from "react";
+import React, { useCallback } from "react";
 
 import * as katex from "katex";
 import "katex/dist/katex.min.css";
 
 import "./Latex.css";
 
-export default function Latex(props) {
-  const ref = useRef();
-  const { inline, latex } = props;
+interface LatexProps {
+  inline?: boolean;
+  latex: string;
+}
 
-  useEffect(() => {
-    if (latex) {
-      katex.render(latex, ref.current, {
-        throwOnError: false,
-      });
-    }
-  }, [latex]);
+export default function Latex({ inline, latex }: LatexProps) {
+  const initLatex = useCallback(
+    (ref: HTMLElement | null) => {
```

Comment:
interesting usage of ref callback here. I haven't seen this before. Is it guaranteed to be called at the right time always? I usually do/see `useRef` and `useEffect` combo.

Reply:
yeah, I usually use useEffect+useRef, but I decided to look it up and per [this Stack Overflow post](https://stackoverflow.com/a/60476525) this should work.

In any case, our katex call depends on the DOM element existing which will cause a re-render and populate ref.current anyway, so I'm under the impression that either will work just fine.

Reply:
nice, thanks for sharing the link.